src/components/Dashboard/TodoList.vue

```diff
@@ -658,6 +659,15 @@ export default class TodoList extends Mixins(DateMixin, EnumMixin, FilingMixin,
     return this.requiresAlteration ? 'Alter Now' : 'Resume'
   }
 
+  /** Whether the File Annual Report button should be disabled. */
+  private isFileAnnualReportDisabled (item: TodoItemIF, index: number): boolean {
+    return (
+      !item.enabled ||
+      (this.isBComp && !this.enableCheckbox[index]) ||
+      !this.isAllowed(AllowableActions.FILE_ANNUAL_REPORT)
+    )
```

Comment:
Pretty much the same logic that was in the template except for `isAllowed()` (and `isCoaPending` is now part of `isAllowed` for this action).